forge-gui-desktop/src/test/java/forge/gamesimulationtests/util/IntegerConstraint.java

```diff
@@ -1,31 +1,94 @@
 package forge.gamesimulationtests.util;
 
-public class IntegerConstraint {
-	public static final IntegerConstraint ONE = new IntegerConstraint( 1 );
-	public static final IntegerConstraint ZERO_OR_MORE = new IntegerConstraint( 0, Integer.MAX_VALUE );
-	public static final IntegerConstraint ZERO_OR_ONE = new IntegerConstraint( 0, 1 );
+import java.util.Objects;
+
+/**
+ * Represents a constraint on an integer value, ensuring it falls within a specified range.
+ * This class enforces proper encapsulation and validation of its internal state.
+ */
+public class IntegerConstraint extends Constraint<Integer> {
+	private static final int MIN_DEFAULT = 0;
+	private static final int MAX_DEFAULT = Integer.MAX_VALUE;
+	
+	// Factory methods for common constraint types
+	public static final IntegerConstraint ONE = new IntegerConstraint(1);
+	public static final IntegerConstraint ZERO_OR_MORE = new IntegerConstraint(0, Integer.MAX_VALUE);
+	public static final IntegerConstraint ZERO_OR_ONE = new IntegerConstraint(0, 1);
 	
-	private final int min;
-	private final int max;
+	/**
+	 * Creates a new IntegerConstraint with default range (0 to Integer.MAX_VALUE)
+	 */
+	public IntegerConstraint() {
+		this(MIN_DEFAULT, MAX_DEFAULT);
+	}
 	
-	public IntegerConstraint( int value ) {
-		this( value, value );
+	/**
+	 * Creates a new IntegerConstraint with specified minimum value
+	 * @param min The minimum allowed value
+	 * @throws IllegalArgumentException if min is negative
+	 */
+	public IntegerConstraint(int min) {
+		this(min, MAX_DEFAULT);
```

Comment:
Are you intentionally changing this from Min -> Min, to Min -> Max? Because its unclear why this change occurred.